src/nvim/diff.c

```diff
@@ -2471,7 +2454,6 @@ bool diff_infold(win_T *wp, linenr_T lnum)
 void nv_diffgetput(bool put, size_t count)
 {
   exarg_T ea;
-  char buf[30];
```

Comment:
This cannot be moved down. It will be used by `ex_diffgetput()`.

Reply:
Sorry, can you elaborate a little bit? `buf` isn't an argument to `ex_diffgetput()`, I don't understand how it can be used there.

Reply:
`ea->arg` is.

Reply:
Moving this down will cause accessing `eap->arg` in `ex_diffgetput()` to be undefined behavior.

Reply:
Hmm, so `ea.arg` is pointing to `buf`, but `buf` is deallocated after the `else` block making `ea.arg` point to the address that's been deallocated...or something. Pointers are confusing :/

OK, reverting. Thanks for the help.

Reply:
It's usually not "deallocated" by most compilers, but it should be undefined behavior according to C standard.